catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/DataListFragment.java

```diff
@@ -350,6 +361,209 @@ public void onItemLongClick(final UserData item, CheckableVH holder) {
 				.show();
 	}
 
+	@VisibleForTesting
+	public void deleteItemsNotInUse(List<UserData> selectedItems) {
+		Project project = projectManager.getCurrentProject();
+		List<UserVariable> globalVariables = project.getUserVariables();
+		List<UserList> globalLists = project.getUserLists();
+		Scene currentScene = projectManager.getCurrentlyEditedScene();
+
+		List<UserData> itemsToDelete = filterDeletableUserDataItems(selectedItems, globalVariables, globalLists, currentScene);
+
+		if (itemsToDelete.size() > 0) {
+			deleteItems(itemsToDelete);
+		}
+	}
+
+	private List<UserData> filterDeletableUserDataItems(List<UserData> selectedItems,
+			List<UserVariable> globalVariables, List<UserList> globalLists, Scene currentScene) {
+		List<UserData> itemsToDelete = new ArrayList<>();
+
+		for (UserData item : selectedItems) {
+			if (globalVariables.contains(item) || globalLists.contains(item)) {
+				if (!checkIfUserDataItemInUse(item, null) && !checkIfUserDataItemInDropDown(item, null)) {
+					itemsToDelete.add(item);
+				}
+			} else {
+				if (!checkIfUserDataItemInUse(item, currentScene) && !checkIfUserDataItemInDropDown(item, currentScene)) {
+					itemsToDelete.add(item);
+				}
+			}
+		}
+
+		return itemsToDelete;
+	}
+
+	private boolean checkIfUserDataItemInUse(UserData item, @Nullable Scene currentScene) {
+		List<Scene> sceneList;
+		Project project = projectManager.getCurrentProject();
+		Sprite currentSprite = projectManager.getCurrentSprite();
+
+		if (currentScene == null) {
+			sceneList = project.getSceneList();
+		} else {
+			sceneList = new ArrayList<>();
+			sceneList.add(currentScene);
+		}
+
+		for (Scene scene : sceneList) {
```

Comment:
As discussed with @gPathpp : Please try to figure out if there are any existing functions for fetching your wanted data. Elsewise please try to rewrite these part of code. There are too many nestings and its not clear how to follow the logic of the function. You could try to split parts of the logic in different functions.